library/src/main/java/com/getstream/sdk/chat/utils/MessageInputController.java

```diff
@@ -46,40 +44,54 @@
     private ChannelViewModel viewModel;
     private Channel channel;
     private MessageInputStyle style;
-    private MediaAttachmentAdapter mediaAttachmentAdapter = null;
-    private MediaAttachmentSelectedAdapter selectedMediaAttachmentAdapter = null;
+    private MediaAttachmentAdapter mediaAttachmentAdapter;
+    private MediaAttachmentSelectedAdapter selectedMediaAttachmentAdapter;
+    private AttachmentListAdapter fileAttachmentAdapter;
+    private AttachmentListAdapter selectedFileAttachmentAdapter;
     private CommandMentionListItemAdapter<MessageInputStyle> commandMentionListItemAdapter;
     private List<Object> commands = null;
     private Context context;
     private StreamViewMessageInputBinding binding;
-    private AttachmentListAdapter fileAttachmentAdapter = null;
-    private AttachmentListAdapter selectedFileAttachmentAdapter = null;
+
     private MessageInputType messageInputType;
-    private List<Attachment> selectedAttachments = null;
+    private List<Attachment> selectedAttachments;
     private MessageInputView.AttachmentListener attachmentListener;
-    private boolean uploadingFile = false;
 
+    private List<Attachment> localAttachments;
+    private UploadManager uploadManager;
     // region Attachment
 
-    public MessageInputController(Context context,
-                                  StreamViewMessageInputBinding binding,
-                                  ChannelViewModel viewModel,
-                                  MessageInputStyle style,
-                                  MessageInputView.AttachmentListener attachmentListener) {
+    public MessageInputController(@NonNull Context context,
+                                  @NonNull StreamViewMessageInputBinding binding,
+                                  @NonNull ChannelViewModel viewModel,
+                                  @NonNull MessageInputStyle style,
+                                  @Nullable MessageInputView.AttachmentListener attachmentListener) {
         this.context = context;
         this.binding = binding;
         this.viewModel = viewModel;
         this.channel = viewModel.getChannel();
         this.style = style;
         this.attachmentListener = attachmentListener;
+        uploadManager = new UploadManager(channel, context);
     }
 
     public List<Attachment> getSelectedAttachments() {
         return selectedAttachments;
     }
 
-    public boolean isUploadingFile() {
-        return uploadingFile;
+    private void updateSelectedAttachment(Attachment attachment, boolean add){
+        if (add){
+            if (selectedAttachments == null)
+                selectedAttachments = new ArrayList<>();
+            selectedAttachments.add(attachment);
+        }else{
+            selectedAttachments.remove(attachment);
+        }
+    }
+
+    public boolean isUploadingFile(){
+        if (uploadManager == null) return false;
```

Comment:
This null check is not needed. `uploadManager` is initialised in constructor. And if you mark `uploadManager` field as final then IDE will not even shoe NPE warning.